File: VB/DiagnosticTool/DignosticController/DefaultFileController.vb
Imports System.IO
Imports System.Runtime.InteropServices

Namespace DashboardDiagnosticTool

    Friend Class DefaultFileController
        Implements IFileController

        Private fileExtension As String = ".xml"

        Public Function TryOpenFile(<Out> ByRef openName As String, ByVal Optional fileName As String = "") As Boolean Implements IFileController.TryOpenFile
            If Equals(Path.GetExtension(fileName), fileExtension) Then
                openName = fileName
                Return True
            End If

            openName = ""
            Return False
        End Function

        Public Function TrySaveFile(<Out> ByRef outFileName As String, ByVal Optional fileName As String = "") As Boolean Implements IFileController.TrySaveFile
            If Equals(Path.GetExtension(fileName), fileExtension) Then
                outFileName = fileName
                Return True
            End If

            outFileName = ""
            Return False
        End Function
    End Class
End Namespace

File: CS/DiagnosticTool/DignosticController/DefaultFileController.cs
using System.IO;

namespace DashboardDiagnosticTool {
    class DefaultFileController : IFileController {
        string fileExtension = ".xml";
        public bool TryOpenFile(out string openName, string fileName = "") {
            if(Path.GetExtension(fileName) == fileExtension) {
                openName = fileName;
                return true;
            }
            openName = "";
            return false;
        }

        public bool TrySaveFile(out string outFileName, string fileName = "") {
            if(Path.GetExtension(fileName) == fileExtension) {
                outFileName = fileName;
                return true;
            }
            outFileName = "";
            return false;
        }
    }
}
